### DISCUSSION TITLE
How to integrate with NextJS 13?

### DISCUSSION THREAD

**Discussion by tspecht** [chakra-ui/panda#857] [Q&A] [answered]
I'm on NextJS 13.4.7. I followed the installation guide, but none of my styles actually end up in the DOM. I believe the issue is that NextJS 13 requires the use of a `StyleRegistry`, but I'm unsure where to get the fully rendered CSS from. Any suggestions?

**Comment by anubra266** [comment]
@tspecht if you inspect the dom, are the atomic classnames generated?

**Comment by tspecht** [comment]
@anubra266 they do, yes, but I can not find any of the actual CSS as part of the DOM. For reference, I'm migrating from `stitches`, where I [needed to use a Registry](https://nextjs.org/docs/app/building-your-application/styling/css-in-js) to get the CSS to be injected. Is there a function provided similar to `getCssText` in stitches?

```

"use client";

import React, { useState } from "react";
import { useServerInsertedHTML } from "next/navigation";
import { getCssText } from "stitches.config";

export default function StitchesRegistry({ children }: { children: React.ReactNode }) {
  const [isRendered, setIsRendered] = useState(false);

  useServerInsertedHTML(() => {
    if (!isRendered) {
      setIsRendered(true);
      return <style id="stitches" dangerouslySetInnerHTML={{ __html: getCssText() }} />;
    }
  });

  return <>{children}</>;
}
```

**Reply by anubra266** [reply]
@tspecht are you sure you imported your `global` which contains the layers in your app?
https://panda-css.com/docs/getting-started/nextjs#import-the-entry-css-in-your-app
![CleanShot 2023-06-29 at 18 11 28@2x](https://github.com/chakra-ui/panda/assets/30869823/301e3e58-ff80-4685-8824-396036f8c9eb)

**Reply by tspecht** [reply]
I am, as I have `tailwind` imported there as well, which works without any issues. My full `globals.css` is 

```
@layer reset, base, tokens, recipes, utilities;

@tailwind base;
@tailwind components;
@tailwind utilities;
```

**Reply by anubra266** [reply]
OKay @tspecht Might need an online reproduction. Try if you can reproduce in stackblitz. Cause I'm unable to reproduce the issue.

**Reply by tspecht** [reply]
@anubra266 here you go! https://stackblitz.com/edit/stackblitz-starters-cexpav

**Comment by asterikx** [comment]
I'm also experiencing weird issues, with styles _sometimes_ not being loaded (classes are present in the DOM). In my case, it's difficult to reporduce. It sometimes happens when refreshing the page with dev tools opened (some caching issues maybe?).

**Comment by anubra266** [comment, marked as answer]
@tspecht you can't have two post css config files. Your `.js` config is used, and you haven't added panda plugin in that config.
Also, your example doesn't include the layers.
And lastly, it seems tailwind isn't letting panda work while it's present.
